details/operator-sdk/api/v1/astraconnector_validator.go

```diff
@@ -39,3 +52,93 @@ func (ai *AstraConnector) ValidateNamespace() *field.Error {
 	}
 	return nil
 }
+
+// ValidateTokenAndAccountID Validates the token and AccoundID provided that AstraConnector should be deployed to.
+func (ai *AstraConnector) ValidateTokenAndAccountID(httpClient util.HTTPClient) *field.Error {
+	cloudBridgeJsonField := util.GetJSONFieldName(&ai.Spec.NatsSyncClient, &ai.Spec.NatsSyncClient.CloudBridgeURL)
+	tokenRefBridgeJsonField := util.GetJSONFieldName(&ai.Spec.Astra, &ai.Spec.Astra.TokenRef)
+	accountJsonField := util.GetJSONFieldName(&ai.Spec.Astra, &ai.Spec.Astra.AccountId)
+	astraHost := getAstraHostURL(ai.Spec.NatsSyncClient.CloudBridgeURL)
+	accountId := ai.Spec.Astra.AccountId
+
+	// Account needs to be a valid UUID
+	_, err := uuid.Parse(accountId)
+	if err != nil {
+		println("Please check account id provided.. Token needs to be UUID")
+		return field.Invalid(field.NewPath(accountJsonField), ai.Name, "Account not valid")
+	}
+
+	config, _ := ctrl.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(config)
+	apiToken, err := getSecret(clientset, ai.Spec.Astra.TokenRef, ai.ObjectMeta.Namespace)
+	if err != nil {
+		log.Info("Check TokenRef, make sure Kubernetes secret was created.")
+		return field.NotFound(field.NewPath(tokenRefBridgeJsonField), ai.Name)
+	}
+
+	if httpClient == nil {
+		httpClient, err = util.SetHttpClient(ai.Spec.Astra.SkipTLSValidation,
+			astraHost, ai.Spec.NatsSyncClient.HostAliasIP, log)
+		if err != nil {
+			log.Info(fmt.Sprintf("invalid cloudBridgeURL provided: %s, format - https://hostname", ai.Spec.NatsSyncClient.CloudBridgeURL))
+			return field.Invalid(field.NewPath(cloudBridgeJsonField), ai.Name, "CloudBridgeURL invalid format")
+		}
+	}
+
+	url := fmt.Sprintf("%s/accounts/%s", astraHost, accountId)
+
+	headerMap := util.HeaderMap{Authorization: fmt.Sprintf("Bearer %s", apiToken)}
+	response, err, cancel := util.DoRequest(context.Background(), httpClient, http.MethodGet, url, nil, headerMap)
+	defer cancel()
+
+	var dnsError *net.DNSError
+	if errors.As(err, &dnsError) {
+		log.Info("Please check CloudBridgeURL provided")
+		return field.Invalid(field.NewPath(cloudBridgeJsonField), ai.Name, "CloudBridgeURL not reachable")
+	}
+
+	// We got a 200 from the GET Account Looks good! no errors
+	if response.StatusCode == 200 {
+		return nil
+	}
+
+	// error handling below
+	if response.StatusCode == 401 {
```

Comment:
are these the only possible error codes? we might want to add a catchall as well.